include/parser.php

```diff
@@ -789,6 +790,7 @@ function do_bbcode($text, $is_signature = false)
 	$replace[] = '<em>$1</em>';
 	$replace[] = '<span style="color: $1">$2</span>';
 	$replace[] = '</p><h5>$1</h5><p>';
+	$replace[] = '<p dir="RTL" align="right">$1</p>';
```

Comment:
I think the better approach is the following:
- avoid hard-coding `align`, either to right of left in html (and in this particular case, not even in css)
- add `dir="auto"` to all heading and paragraph tags so if the text starts with RTL related character, the browser will understand what to do with it.

Reply:
It may be better but is it actually practical - those sound like very substantial changes and we to keep out customisations as small as possible.

Reply:
1. i think when we create and call a new bbcode tag 'rtl' it should be hard-coded to right. because it supposed to be RTL, not auto. and yes if we could do it with CSS not directly in HTML that would be a better approach.
2. adding dir="auto" to all headings and paragraphs is a better approach because this way user don't need to use a bbcode tag each time writing a post, but it will be forum wide. this suggestion could be in another commit which i cant make because of lack of knowledge.

Reply:
As per my understanding, we should specify RTL or LTR when we know the content will be RTL or LTR all the time. in social networks, messengers,... in which we have people with different languages, RTL and LTR should not be forced to any element which might include RTL or LTR text.

With the same philosophy, I am contributing to Plume (a Free and Open Source federated blogging platform) to support RTL and LTR natively.

Reply:
> It may be better but is it actually practical - those sound like very substantial changes and we to keep out customisations as small as possible.

The change I am suggesting is not going to harm anything. All the posts remain untouched. All the people who post using English or any other LTR languages, face no changes. But people using RTL languages, will get more convenience and furthermore, not extra and explicit RTL/LTR code or styling is going to be imposed on posts that later cannot be changes easily.

Of course what I say should apply on whole website (wherever there might be a RTL/LTR text (including title of the thread, latest posts,...) but this can be implemented later in a new dedicated branch

Reply:
I'm referring to the code base here, and the fact that is a fork and we want to minimise the amount of changes we are making locally.

Reply:
I understood. So we should make such changes on the the fluxbb repo (http://github.com/fluxbb/fluxbb). So, my suggestion is to not make any local change which can raise conflict with the upstream repo

Reply:
by the way, my suggestion is not a major one. here you can see the result:

![image](https://user-images.githubusercontent.com/11241315/58628022-dbeb4600-82ed-11e9-9672-fe0eda95233d.png)

As you can see, on right side, change is made. RTL texts are RTL and LTR texts are LTR as expected. For this, we need two changes:
- remove `align:left;` from `Air.css:741`
- add `dir="auto"` to all `p` and maybe all headings in post body.

as simple as this.

If you cannot except this change, then wait for FluxBB update

Reply:
@tomhughes What is your suggestion? If we make changes on FluxBB, I am not sure if you are going to pull it since this repo is not following the new updates on FluxBB. Should we send PR to both repos? both FluxBB and here?

Reply:
Is FluxBB actually alive?

Reply:
Ah there is a new version so I should look at rebasing.

If it's a general good change then I would suggest offering it upstream, though if it's reasonably simple we can carry it locally in the meantime.

Reply:
I don't know. Since you are running an active instance of it, I think you can tell us the best approach to take. As far as I see, there has been some activities on FluxBB in last few months. though, forks are much more active that the main repo.

You tell us what is the best approach to take

Reply:
I plan to keep maintaining FluxBB. :)